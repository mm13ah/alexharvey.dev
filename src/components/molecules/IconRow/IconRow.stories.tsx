import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import {
  SiReact,
  SiStorybook,
  SiJest,
  SiCypress,
  SiTailwindcss,
  SiStyledcomponents,
  SiExpo,
  SiApollographql,
  SiCircleci,
  SiDocker,
  SiExpress,
  SiGithubactions,
  SiGraphql,
  SiNodedotjs,
  SiPostgresql,
  SiPrisma,
  SiSequelize,
} from 'react-icons/si';
import { TbBrandReactNative } from 'react-icons/tb';
import { BsPencilSquare } from 'react-icons/bs';
import { FaGraduationCap } from 'react-icons/fa';
import { IconRow } from './IconRow';

export default {
  title: 'Molecules/IconRow',
  component: IconRow,
  args: {
    color: 'white',
  },
} as Meta<typeof IconRow>;

export const Frontend: StoryFn<typeof IconRow> = (args) => (
  <IconRow {...args} title="Frontend">
    <SiReact size={20} color="white" />
    <SiStorybook size={20} color="white" />
    <SiJest size={20} color="white" />
    <SiCypress size={20} color="white" />
    <SiTailwindcss size={20} color="white" />
    <SiStyledcomponents size={20} color="white" />
    <SiExpo size={20} color="white" />
    <TbBrandReactNative size={20} color="white" />
  </IconRow>
);
Frontend.storyName = 'Frontend';

export const Backend: StoryFn<typeof IconRow> = (args) => (
  <IconRow {...args} title="Backend">
    <SiNodedotjs size={20} color="white" />
    <SiGraphql size={20} color="white" />
    <SiExpress size={20} color="white" />
    <SiPostgresql size={20} color="white" />
    <SiApollographql size={20} color="white" />
    <SiJest size={20} color="white" />
    <SiPrisma size={20} color="white" />
    <SiSequelize size={20} color="white" />
  </IconRow>
);
Backend.storyName = 'Backend';

export const Other: StoryFn<typeof IconRow> = (args) => (
  <IconRow {...args} title="Other">
    <SiCypress size={20} color="white" />
    <SiCircleci size={20} color="white" />
    <SiGithubactions size={20} color="white" />
    <SiDocker size={20} color="white" />
  </IconRow>
);
Other.storyName = 'Other';

export const Writing: StoryFn<typeof IconRow> = (args) => (
  <IconRow {...args} title="Writing">
    <BsPencilSquare size={20} color="white" />
  </IconRow>
);
Writing.storyName = 'Writing';

export const Education: StoryFn<typeof IconRow> = (args) => (
  <IconRow {...args} title="Education">
    <FaGraduationCap size={20} color="white" />
  </IconRow>
);
Education.storyName = 'Education';
