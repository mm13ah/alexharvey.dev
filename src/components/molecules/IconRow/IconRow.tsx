import React from 'react';
import { SiGraphql, SiNodedotjs, SiReact, SiTypescript } from 'react-icons/si';

interface Props {
  color?: string;
}

export const IconRow = ({ color }: Props) => (
  <div className="flex items-center gap-x-10">
    <SiTypescript size={38} color={color ?? 'white'} />
    <SiReact size={38} color={color ?? 'white'} />
    <SiGraphql size={38} color={color ?? 'white'} />
    <SiNodedotjs size={38} color={color ?? 'white'} />
  </div>
);
