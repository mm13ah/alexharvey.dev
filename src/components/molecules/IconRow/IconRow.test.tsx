import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { IconRow } from './IconRow';

describe('IconRow', () => {
  it('should render', () => {
    render(<IconRow />);
  });
  it('should match a snapshot', () => {
    const tree = renderer.create(<IconRow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
