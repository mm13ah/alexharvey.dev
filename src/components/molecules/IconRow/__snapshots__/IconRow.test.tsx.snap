// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconRow should match a snapshot 1`] = `
<div
  className="flex flex-col sm:flex-row sm:items-center gap-x-5"
>
  <h3
    className="text-white font-medium tracking-wider text-2xl"
  />
  <div
    className="flex gap-x-5 my-2 sm:my-0"
  >
    <span />
  </div>
</div>
`;
